Source/Core/Engine/ResourceSystem.hpp

```diff
@@ -75,6 +75,14 @@ class ResourceSystem : public EventObject
   // Resources that were modified in the editor.
   HashSet<ResourceId> mModifiedResources;
 
+  // Map of core resource library names to their respective resource libraries
+  typedef OrderedHashMap<String, ResourceLibrary*> LoadedSetMap;
+  LoadedSetMap LoadedPlasmaCoreLibraries;
```

Comment:
Is it worth having both LoadedPlasmaCoreLibraries and LoadedDependencyLibraries, given that right now they seem to do the exact same thing? If they are going to have different roles in the future, then the comments in this file should be updated to reflect that.

Reply:
I can remove it for now, we definitely need to go back and look at this whole section of code though and do some major design changes for loading on demand and only dependent libraries.